lib/CMakeLists.txt

```
@@ -99,6 +99,17 @@ if (MSVC)
     target_compile_definitions(SoapySDR PUBLIC -DNOMINMAX) #enables std::min and std::max
 endif ()
 
+#windows header check
+include(CheckIncludeFiles)
+CHECK_INCLUDE_FILES(windows.h HAS_WINDOWS_H)
+
+#check missing vasprintf on windows
+include(CheckFunctionExists)
+check_function_exists(vasprintf HAS_VASPRINTF)
```

```suggestion
check_symbol_exists(vasprintf "stdio.h" HAS_VASPRINTF)
```
The `check_function_exists` macro does not work here, because it checks link-level symbol existence. On mingw, these symbols exist in the crt, but are not available in the header unless `_GNU_SOURCE` is set, since this function is a GNU extension. The same is true on non-mingw platforms also, but there the compiler often tends to set `_GNU_SOURCE` itself as part of the platform ABI.

---

turns out that didnt work on one ubuntu20.04, so I pushed a change to use vsnprintf https://github.com/pothosware/SoapySDR/pull/319/commits/68b3917831599e1f31e9425237d5f18cf255ff52

---

Just using the std function sounds good to me.